app/src/main/java/com/github/ybecker/epforuml/sensor/CameraActivity.kt

```diff
@@ -116,6 +159,84 @@ class CameraActivity : AppCompatActivity() {
 
     }
 
+    private fun startRecordingVideo() {
+
+        val videoCapture = this.videoCapture ?: return
+
+        isRecording=true
+
+        val curRecording = recording
+        if (curRecording != null) {
+            curRecording.stop()
+            recording = null
+            return
+        }
+
+        val name = SimpleDateFormat(FILENAME_FORMAT, Locale.US)
+            .format(System.currentTimeMillis())
+        val contentValues = ContentValues().apply {
+            put(MediaStore.MediaColumns.DISPLAY_NAME, name)
+            put(MediaStore.MediaColumns.MIME_TYPE, "video/mp4")
+            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P) {
+                put(MediaStore.Video.Media.RELATIVE_PATH, "Movies/CameraX-Video")
+            }
+        }
+
+        val context = this
+        val mediaStoreOutputOptions = MediaStoreOutputOptions
+            .Builder(contentResolver, MediaStore.Video.Media.EXTERNAL_CONTENT_URI)
+            .setContentValues(contentValues)
+            .build()
+
+        // launch a new Recording
+        recording = videoCapture.output
+            .prepareRecording(this, mediaStoreOutputOptions)
+            .apply {
+                if (PermissionChecker.checkSelfPermission(
+                        context,
+                        Manifest.permission.RECORD_AUDIO
+                    ) ==
+                    PermissionChecker.PERMISSION_GRANTED
+                ) {
+                    withAudioEnabled()
+                }
+            // start the new recording
+            }.start(ContextCompat.getMainExecutor(context)) { recordEvent ->
+            when (recordEvent) {
+                is VideoRecordEvent.Start -> {
+                    isRecording=true
+                    Log.d(TAG, "Start recording")
+
+                }
+                //when the video ends
+                is VideoRecordEvent.Finalize -> {
+                    isRecording=false
+                    val uri = recordEvent.outputResults.outputUri
+                    //when no errors the video is saved correctly, we can go back the the question
+                    if (!recordEvent.hasError()) {
+                        val msg = "Video capture succeeded: " +
+                                "$uri"
+                        goBackToQuestion(uri.toString())
+                        Log.d(TAG, msg)
+                    //if there is an error stop recording, and stay on the activity so the user can retake a video
+                    } else {
+                        recording?.close()
+                        recording = null
+                        Toast.makeText(this, "Error during the recording", LENGTH_SHORT)

```

Comment:
As warned by the CI, your toast won't display as you didn't call the .show() method on the toast

Reply:
Thank you !